src/terrain/services/filesystem/stat.clj

```diff
@@ -67,6 +69,17 @@
   (assoc stat-map
          :label (paths/id->label user path)))
 
+(defn get-public-data-user
+  "Returns the anonymous user for public data if a user is not provided"
+  [user paths]
+  (let [paths (if (sequential? paths) paths [paths])
+        has-private-paths? (some #(not (string/starts-with? % (cfg/fs-community-data))) paths)
+        request-user (if has-private-paths?
+                       user
+                       (or user "anonymous"))]
+    (or request-user (throw+ {:type :clojure-commons.exception/not-authorized
+                              :user user}))))
```

Comment:
Just to make the other comment I made easier, here's what I'd do here:

```suggestion
(defn get-public-data-user
  "Returns the anonymous user for public data if a user is not provided"
  ([user paths ids]
   (let [paths (if (sequential? paths) paths [paths])
         has-private-paths? (or (seq ids) (some #(not (string/starts-with? % (cfg/fs-community-data))) paths))
         request-user (if has-private-paths?
                        user
                        (or user "anonymous"))]
     (or request-user (throw+ {:type :clojure-commons.exception/not-authorized
                               :user user}))))
  ([user paths]
   (get-public-data-user user paths nil))
```

untested code so make sure I didn't make a typo, but anyway :smile: